Migrate Settings from Earlier Versions

papeh opened the pull request:
<!-- Reviewable:start -->
This change is [<img src="https://reviewable.io/review_button.svg" height="34" align="absmiddle" alt="Reviewable"/>](https://reviewable.io/reviews/sillsdev/flexbridge/345)
<!-- Reviewable:end -->

papeh committed:
Migrate Settings from Earlier Versions (#343)

* Test that all settings use CrossPlatformSettingsProvider
* Migrate the latest settings from previous versions that
  did not use CrossPlatformSettingsProvider
* Remove settings from projects that don't use them

papeh committed:
Migrate Settings from Earlier Versions (#343) Followup

* Switch to our in-house file utilities for tests
  to avoid an indirect dependency on System.Windows.Forms

jasonleenaylor reviewed (commented):
> Reviewed 22 of 39 files at r1, all commit messages.
> *[Reviewable](https://reviewable.io/reviews/sillsdev/flexbridge/345)* status: 22 of 39 files reviewed, all discussions resolved (waiting on @jasonleenaylor)



<!-- Sent from Reviewable.io -->

jasonleenaylor reviewed (approved):
> Reviewed 15 of 39 files at r1, 7 of 7 files at r2, 4 of 4 files at r3, all commit messages.
> *[Reviewable](https://reviewable.io/reviews/sillsdev/flexbridge/345)* status: :shipit: complete! all files reviewed, all discussions resolved (waiting on @papeh)



<!-- Sent from Reviewable.io -->

papeh committed:
Really remove System.Windows.Forms reference

As it happens, the library was not the culprit.
Also downgrade NUnit, whose newest version doesn't
work for us for some reason.
